UTRannotator.pm

```diff
@@ -0,0 +1,1306 @@
+            #if the variant is entirely in the uORF (within 5'UTR
+            if(exists($existing_ref_uORF{$start_pos})) {
+                my @stops = sort {$a <=> $b} @{$existing_ref_uORF{$start_pos}};
+                $check_point = $stops[0]-1;
+            }else{
+            #if the existing uORF is an oORF
+                $check_point = $utr_length-1;
+            }
+
+            # only check the ones between start and stop codons
+            # ignore the cases at the boundary of start and stop codon since the effect on start/stop would be evaluated anyway
+            next if(($mut_pos<$start_pos+3)|($end_pos>$check_point));
+            #check whether there are new stop codon induced by this mutation
+            my @mut_stops;
+            next unless(exists($mut_uORF{$start_pos}));
+            @mut_stops = sort {$a <=> $b} @{$mut_uORF{$start_pos}};
+
+            my $mut_stop = $mut_stops[0];
+            if($mut_stop<$check_point){
+                $flag=1;}
+
+            if($flag){
+
+                    #getting the Kozak context and Kozak strength of the start codon
+
+                    if ((($start_pos-3)>=0)&&($sequence[($start_pos+3)])){
+                        $current_kozak = $sequence[($start_pos-3)].$sequence[($start_pos-2)].$sequence[$start_pos-1]."ATG".$sequence[$start_pos+3];
+                    }
+                    else{
+                        $current_kozak = '-';
+                    }
+
+                    if ($current_kozak !~ /-/){
+                        my @split_kozak = split //, $current_kozak;
+                        $current_kozak_strength = 1;
+                        if ((($split_kozak[0] eq 'A')||($split_kozak[0] eq 'G'))&&($split_kozak[6] eq 'G')){
+                            $current_kozak_strength = 3;
+                        }
+                        elsif ((($split_kozak[0] eq 'A')||($split_kozak[0] eq 'G'))||($split_kozak[6] eq 'G')){
+                            $current_kozak_strength = 2;
+                        }
+                    }
+
+                    $uSTOP_gained_KozakContext=$current_kozak;
+                    $uSTOP_gained_KozakStrength=$kozak_strength{$current_kozak_strength}? $kozak_strength{$current_kozak_strength}:$current_kozak_strength;
+
+
+                    #the annotation of the original uORF
+                    if (exists($existing_ref_uORF{$start_pos})){ #if there is stop codon within 5'UTR
+                        $uSTOP_gained_ref_type = "uORF"
+                    } elsif (($utr_length-$start_pos) % 3){
+                        $uSTOP_gained_ref_type = "OutOfFrame_oORF";
+                    } else {
+                        $uSTOP_gained_ref_type = "InFrame_oORF";
+                    }
+
+                    $uSTOP_gained_ref_StartDistanceToCDS = $utr_length - $start_pos;
+                    $uSTOP_gained_newSTOPDistanceToCDS = $mut_utr_length - $mut_stop;
+
+                    #find evidence from sorfs.org
```

Comment:
It would be clear if the comment said that the evidence is fetched from the file (the file is the one fetched from sorfs.org).

Reply:
Refurbished a little bit.